### PR TITLE
Change licence to Apache 2

### PR DESCRIPTION
**Pull request by itowlson**
The Kubernetes extension is becoming a CNCF project.  As part of this, the CNCF requires us to change the licence to Apache 2.0.

This issue asks our contributors, who contributed their work under the MIT licence, to consent to the project being relicensed as Apache 2.0.

Thanks to all of you!

@apupier 
@itowlson 
@pantosha 
@flowftw 
@flynnduism 
@evidolob 
@lstocchi 
@ivolzhevbt 
@brendandburns 
@rakeshvanga 
@ezramorris 
@markszente 
@shreyaskarnik 
@Persistent13 
@bacongobbler 
@amsoedal 
@Techmaster21 
@dev01d 
@Tatsinnit 
@bnookala 
@jo-goro 
@bfitzpat 
@ltsbkuster 
@aelij 
@mattjanson 
@gorkem 
@hermanho 
@devlead 
@tedchamb 
@amirschw 
@remcohaszing 
@ahmadalli 
@mabouchacra 
@antonyveyre 
@ederst 
@alexkreidler 
@medzin
@azatsarynnyy
@khvn26 
@meeech 
@Flanker32 
@yehiyam 
@oik741 
@dgkanatsios 
@mccrackend 
@andxu 
@squillace 
@mostafahussein 
@AntoineGa 
@TheBay0r 
@matlec 
@SeriousM 
@ubikusss 
@tomasr 
@kameshsampath 
@testforstephen 
@sozercan 
@radu-matei 
@karataliu 
@evanlouie 
@arehmandev 
@technosophos

### CODE REVIEW

**Review by devlead — APPROVED** [review]

**Comment by squillace** [comment]
Hi all. One of the "joys" of contributing a repo to the CNCF is the conversion of any licenses. We apologize for the hassle, but I will be pinging you all gently so that we can make the appropriate change and keep on keeping on. If anyone has any issues with this, please do not hesitate to reach out.

**Comment by squillace** [comment]
👍 for me. officially.

**Review by dgkanatsios — APPROVED** [review]

**Comment by khvn26** [comment]
👍

**Comment by dgkanatsios** [comment]
Ship it!

**Review by amirschw — APPROVED** [review]
🚀

**Comment by hermanho** [comment]
👍

**Review by bnookala — APPROVED** [review]

**Review by lstocchi — APPROVED** [review]

**Comment by mattjanson** [comment]
👍 Fine for me

**Comment by flowftw** [comment]
Fine for me as well. 👌

**Review by remcohaszing — APPROVED** [review]

**Review by TheBay0r — APPROVED** [review]

**Comment by meeech** [comment]
👍🏼

**Review by flowftw — APPROVED** [review]

**Review by gorkem — APPROVED** [review]

**Comment by ltsbkuster** [comment]
👍

**Review by markszente — APPROVED** [review]

**Review by ezramorris — APPROVED** [review]

**Review by alexkreidler — APPROVED** [review]

**Comment by medzin** [comment]
LGTM!

**Review by jo-goro — APPROVED** [review]

**Comment by dev01d** [comment]
🎉👍

**Review by Techmaster21 — APPROVED** [review]

**Review by technosophos — APPROVED** [review]
🐾 👍

**Comment by aelij** [comment]
👍

**Review by azatsarynnyy — APPROVED** [review]

**Review by apupier — APPROVED** [review]

**Review by bacongobbler — APPROVED** [review]

**Review by mabouchacra — APPROVED** [review]

**Comment by radu-matei** [comment]
LGTM!

**Review by ivolzhevbt — APPROVED** [review]

**Comment by yehiyam** [comment]
LGTM

**Review by yehiyam — APPROVED** [review]

**Review by radu-matei — APPROVED** [review]

**Review by evidolob — APPROVED** [review]

**Comment by amsoedal** [comment]
👍👍

**Comment by pantosha** [comment]
👍

**Review by mattjanson — APPROVED** [review]

**Review by ahmadalli — APPROVED** [review]

**Review by Tatsinnit — APPROVED** [review]
/lgtm ❤️☕️

**Review by Persistent13 — APPROVED** [review]

**Review by ltsbkuster — APPROVED** [review]

**Comment by itowlson** [comment]
@paroga I need to add you to this list because of just merging your PR - could you approve the change of licence please?  Thanks and sorry for the faff!

**Review by paroga — APPROVED** [review]

**Review by shreyaskarnik — APPROVED** [review]

**Comment by lstocchi** [comment]
Me and @apupier talked with our legal team. They confirmed Red Hat gave Microsoft a license that would allow Microsoft to do the relicensing without needing any further agreement, each Red Hatters is covered and automatically approves this PR.
With this message @bfitzpat, who is the last Red Hatter left to approve it, approves this license change. Thank you.

**Comment by squillace** [comment]
OK, more effort at polite harassment: 

@AntoineGa and @andxu, we need your approval to modify the license as we donate this to the CNCF.

We also need the approvals of:

@antonyveyre
@arehmandev
@ederst
@evanlouie
@Flanker32
@kameshsampath
@karataliu
@matlec
@mccrackend
@mostafahussein
@oik741
@rakeshvanga
@SeriousM
@tedchamb
@testforstephen
@tomasr
@ubikusss

So we're getting closer!

**Review by tomasr — APPROVED** [review]

**Review by hermanho — APPROVED** [review]

**Review by mostafahussein — APPROVED** [review]

**Review by testforstephen — APPROVED** [review]

**Review by matlec — APPROVED** [review]

**Review by sozercan — APPROVED** [review]

**Review by ederst — APPROVED** [review]
fine for me

**Comment by squillace** [comment]
We're getting there! thanks for everyone in various timezones for weighing in. <3 to all. We got these remaining for those moving into the main portion of the western hemisphere workday:

@andxu
@AntoineGa
@antonyveyre
@arehmandev
@brendandburns
@evanlouie
@Flanker32
@kameshsampath
@karataliu
@mccrackend
@oik741
@rakeshvanga
@SeriousM
@tedchamb
@ubikusss

**Comment by brendandburns** [comment]
LGTM!

**Comment by evanlouie** [comment]
lgtm 👍

**Comment by kameshsampath** [comment]
Looks good to me

**Review by SeriousM — APPROVED** [review]
lgtm 🧡

**Comment by squillace** [comment]
All right! Good afternoon EU and morning USA! I believe we're down to the following. 

LETS DO THIS!!!

@andxu
@AntoineGa
@antonyveyre
@arehmandev
@ederst
@Flanker32
@karataliu
@mccrackend
@oik741
@rakeshvanga
@tedchamb
@ubikusss

**Review by ubikusss — APPROVED** [review]

**Comment by squillace** [comment]
whoops, @ubikusss, I think we got you!

**Review by ederst — APPROVED** [review]
double approve :D

**Comment by rakeshvanga** [comment]
LGTM.

**Comment by squillace** [comment]
We're soooo close. I'm going to start Twitter DMing to catch people shortly. Then I'll go instagram. 

@andxu
@AntoineGa
@antonyveyre
@arehmandev
@Flanker32
@karataliu
@mccrackend
@oik741
@tedchamb

**Comment by squillace** [comment]
OK, all, we're almost there. I'm going to personally reach out. If people have contact info, you can ping me on the vs-code-kubernetes slack in kubernetes, or email at first.last at microsoft.com.

@andxu
@AntoineGa
@antonyveyre
@arehmandev
@Flanker32
@karataliu
@mccrackend
@oik741
@tedchamb

**Review by Flanker32 — APPROVED** [review]

**Comment by lstocchi** [comment]
Ping. Waiting for you guys. 

@andxu
@AntoineGa
@antonyveyre
@arehmandev
@mccrackend
@oik741
@tedchamb

**Review by mccrackend — APPROVED** [review]

**Review by oik741 — APPROVED** [review]
LGTM

**Comment by squillace** [comment]
at this point, I think we have only the following left, and we've reached out multiple times as best we can. I would propose that we were covered by the CLA bot and assert that we should merge the license change. How do we all see it?

@andxu
@AntoineGa
@antonyveyre
@arehmandev
@tedchamb

**Comment by AntoineGa** [comment]
LGTM

**Comment by itowlson** [comment]
Okay, I will merge the new licence unless someone asks not to in the next _rolls dice_ 24 hours.

**Pull request merged**
